src/libraries/System.Runtime.InteropServices/gen/Microsoft.Interop.SourceGeneration/Marshalling/CharMarshaller.cs

```diff
@@ -124,7 +128,8 @@ public IEnumerable<StatementSyntax> Generate(TypePositionInfo info, StubCodeCont
 
         public bool UsesNativeIdentifier(TypePositionInfo info, StubCodeContext context)
         {
-            return context.IsInStubReturnPosition(info) || (info.IsByRef && !context.SingleFrameSpansNativeContext);
+            MarshalDirection elementMarshalDirection = MarshallerHelpers.GetMarshalDirection(info, context);
+            return !IsPinningPathSupported(info, context) && (elementMarshalDirection != MarshalDirection.ManagedToUnmanaged || info.IsByRef);
```

Comment:
Nit: I find it easier to reason about when can we pass the managed identifier to a native parameter without changing anything, and then this method is `not` that, but I'm also fine with whatever makes the most sense to you.
```suggestion
            return !(IsPinningPathSupported(info, context) || (elementMarshalDirection == MarshalDirection.UnmanagedToManaged && !info.IsByRef));
```

Reply:
I'd rather keep it as is as the current condition better matches the code in `Generate` I think.